### PR TITLE
Implement load.

### PR DESCRIPTION
**Pull request by rowillia**
This change isn't as optimized as it could be, it's only 2.5X faster, but wanted to get this reviewed before going further with optimizing.

### CODE REVIEW

**Comment by salami162** [comment]
I read half of it. I can give a half +1. But would like to `jit.py` more give a 100% +1

File: toastedmarshmallow/jit.py [hunk 1]
```diff
@@ -61,15 +54,40 @@ def attr_str(attr_name):
 class FieldSerializer(object):
     """Base class for generating code to serialize a field.
     """
+    def __init__(self, context=None):
+        # type: (JitContext) -> None
```
Comment (chriselion) [inline toastedmarshmallow/jit.py]: nit: Do you need `Optional[JitContext]` here, or does it figure it out from the default arg?

File: tests/test_toasted_marshmallow.py [hunk 2]
```diff
@@ -7,23 +7,51 @@
...
-        key = fields.String()
-        value = fields.Integer()
+        key = fields.String(default='world')
+        value = fields.Integer(missing=42)
     return TestSchema()
 
 
-def test_marshmallow_integration(schema):
-    schema.jit = toastedmarshmallow.Jit
+@pytest.mark.parametrize('jit', [toastedmarshmallow.Jit,
```
Comment (chriselion) [inline tests/test_toasted_marshmallow.py]: General question: is there a no-op jit (or `None`?) that you can test with to make sure the behavior agrees with marshmallow everywhere?

File: toastedmarshmallow/jit.py [hunk 3]
```diff
@@ -278,94 +363,200 @@ def generate_marshall_method_body(schema, on_field, context):
...
+                if not _VALID_IDENTIFIER.match(attr_name):
+                    # If attr_name is not a valid python identifier, it can only
+                    # be accessed via key lookups.
+                    serializer = DictSerializer(context)
+
+                body += serializer.serialize(
                     attr_name, field_symbol, assignment_template, field_obj)
+
+                if not context.is_serializing and field_obj.load_from:
+                    # Marshmallow has a somewhat counter intuitive behavior.
```
Comment (chriselion) [inline toastedmarshmallow/jit.py]: TIL.

Is there a test that checks this against the marshmallow behavior? It looks like `test_generate_unmarshall_method_bodies_with_load_from` is testing the generated code, but if marshmallow ever changes its behavior you might not notice.

**Comment by chriselion** [comment]
I don't feel qualified to review most of this, but left a few general comments.

File: toastedmarshmallow/jit.py (hunk 3 above)
Comment (rowillia) [reply toastedmarshmallow/jit.py]: Running the Marshmallow tests themselves would catch that behavior change.

**Comment by rowillia** [comment]
🚀

**Pull request merged**